Header.c

```diff
@@ -286,10 +288,17 @@ int Header_calculateHeight(Header* this) {
       }
       maxHeight = MAXIMUM(maxHeight, height);
    }
+
+   if (maxHeight == pad) {
+      maxHeight = 0;
+      this->pad = 0;
+   }
```

Comment:
Maybe instead of assigning `this->pad` in different locations use an else branch here:
```c
} else
    this->pad = pad;
```
?